README.md

```diff
@@ -42,3 +42,312 @@ Run linters.
 cargo make clippy
 ```
 
+## Usage
+
+In this section is presented the integration of this library into your application, enabling you to kickstart 
+CQRS/Event sourcing implementation.
+
+### Event sourcing
+
+> Event sourcing is a software architectural pattern used to capture and store the changes or events that occur in an 
+application's state over time. Instead of persisting the current state of an object or entity, event sourcing focuses 
+on recording a sequence of events that have led to that state.
+
+#### Aggregate
+
+To initiate the implementation of CQRS/Event sourcing in your codebase, the initial building block to create is an `Aggregate`.
+
+> In the context of event sourcing, an aggregate is a fundamental concept that represents a cluster of domain objects 
+and their state changes. It is one of the key building blocks used to model and manage data in an event-sourced system.
+
+> An aggregate, therefore, is responsible for processing incoming commands and generating corresponding events that 
+reflect the changes to its state. It encapsulates domain logic and rules related to specific operations on the data it 
+manages. Aggregates provide a clear boundary for consistency and concurrency control within the event-sourced system.
```

Comment:
If this is a quote I would put a `source` if possible

Reply:
Mmmh no source right there.. Should i remove the quotation?

Reply:
If it's not a quote I think having it in plain text is more sensible